### DISCUSSION TITLE
self-loop as a feature for Graph Classification

### DISCUSSION THREAD

**Discussion by Eyemissing** [pyg-team/pytorch_geometric#6358] [Q&A]
In my case, I have a scenario, where a self-loop is an important information used as a feature for my graph classification. As far as I understand this is in GNN not possible. Is this correct? what might be a work-around?
BR Mike

**Comment by rusty1s** [comment]
Most GNN layers will already implicitly assume a self-loop by adding a simple skip-connection. If you want to treat self-loops explicitly, you could do something like
```python
out = conv1(x, edge_index[:, edge_index[0] != edge_index[1]]) + conv2(x, edge_index[:, edge_index[0] == edge_index[1]])
